src/interface-adapters/presenters/problems.presenter.ts

```diff
@@ -0,0 +1,13 @@
+import { Problem } from '@/entities/models/problem';
+
+export function problemsPresenter(problems: Problem[]) {
+    return problems.map((problem) => ({
+        id: problem.id,
+        problem: problem.problem,
+        solution: problem.solution,
+        subject: problem.subject,
+        level: problem.level,
+        methods: problem.methods,
+        title: problem.title,
+    }));
+}
```

Comment:
If we always want the list of problems to be presented the same as singular problems, I think this would be "safer".

```suggestion
import type { Problem } from '@/entities/models/problem';
import { problemPresenter } from './problem.presenter';

export function problemsPresenter(problems: Problem[]) {
    return problems.map(problemPresenter);
}
```